features/storage/TESTS/kvstore/general_tests_phase_1/main.cpp

```diff
@@ -85,6 +85,8 @@ static void kvstore_init()
     res = bd->init();
     TEST_ASSERT_EQUAL_ERROR_CODE(0, res);
     int erase_val = bd->get_erase_value();
+    // Clear out any stale data that might be left from a previous test
+    bd->erase(0, bd->size());
```

Comment:
This is a bit dangerous/time consuming.
On some platforms these tests run against SD card, so clearing the whole card might stale the whole test.
Maybe some sensible value should be used here.

Reply:
Any ideas for how to find a sensible value? For the SecureStore case the test explicitly specifies the amount of storage that it needs via the SlicingBlockDevice's that it creates. But I didn't see anything equivalent for the FSStore or bare TDBStore that would bound the portion of the block device that the kvstore might use.

Reply:
There is no easy answer to this.. could it help if we erase just one erase area?

Reply:
I don't think that is sufficient, particularly on internal flash with small block sizes where that might just erase 512 bytes. 
Is the size calculation used for the SlicingBlockDevice roughly applicable to the other tests as well? If so I can try using that as a reasonable baseline.
(In my testing, erasing the two SecureStore SlicingBlockDevice instances was sufficient to resolve the failure I was seeing; I made the erase more generic in an attempt to address the class of problem rather than just one instance).

Reply:
Pushed up a new version that uses the SlicingBlockDevice size as an estimate for how much to erase (and rebased on latest master while I was at it).